### ISSUE TITLE
Complete mess for power companies and grid operators in Norway

### ISSUE THREAD

**Issue by Gazer75** [osmlab/name-suggestion-index#8021]
This index is a mess for these. Mixing power company selling or producing energy with the subsidiary grid operator that only deals with lines and substations.
Not sure I like this index as it can cause more harm than good as it is.

The list of errors is just to long and I'm not going to list them all.

**Comment by 1ec5** [comment]
Thanks for letting us know that there’s a problem. Most of these entries started out from analyzing the most common values and tagging combinations in the OSM database for certain feature types, so it could also reflect an underlying data issue too.

Since you’re familiar with power suppliers in Norway, could you give an example or two of the kind of mixups you’re seeing?

**Comment by Gazer75** [comment]
Its a bit of a mess currently and not finished.
Due to regulations in Europe the monopoly part of grid operations had to be separated from the parent company.
I've been working on updating operator names, but there is a lot to keep track of.

Example:
BKK changed its name to Eviny and retained the old BKK for its grid operator subsidiary. But initially it used the old "BKK Nett" and recently renamed it to just "BKK". (Not updated in OSM to reflect this yet).
At the same time the operations of power stations is done by Eviny Fornybar, another subsidiary. Because here a producer can't sell directly to end users. Eviny Marked is selling to end customers.
See the mess this is? :) It's the same for all these energy companies.

A wikidata mach for grid operators has to look at the industry (P452) and find Q62058278 or Q472093 unless its the TSO Statnett which get Q112046.
There are some errors in wikidata where DSOs have gotten the Q112046 (TSO) value as well. I've fixed that when I see it.
If the above is true then its a suggestion for power=line|minor_line|substation, items that would be part of this.

There are exceptions for pure production radials where the operator of a power station also operates these. Like a line from power station to a nearby substation or a line from station to the dam up on the mountain to control gates and such.

There are also some companies that merged their grid operation and still retained parents which may lead to some confusion.

How often is wikidata checked for updates, or is this all done manually here?

**Comment by 1ec5** [comment]
> There are also some companies that merged their grid operation and still retained parents which may lead to some confusion.

Yes, these situations tend to cause a mess in Wikidata too, because Wikipedia tends to conflate multiple topics in the same article for convenience. But Wikidata does want to accurately reflect the organizational structure in a manner very reminiscent of how we micromap things in OSM.

> How often is wikidata checked for updates, or is this all done manually here?

I think that’s whenever @tas50 runs https://github.com/ideditor/nsi-collector/ at this point.

**Comment by Gazer75** [comment]
> > How often is wikidata checked for updates, or is this all done manually here?
> 
> I think that’s whenever @tas50 runs https://github.com/ideditor/nsi-collector/ at this point.
Will names that no longer exist be removed from the database?

**Comment by LaoshuBaby** [comment]
> > How often is wikidata checked for updates, or is this all done manually here?
> 
> I think that’s whenever @tas50 runs https://github.com/ideditor/nsi-collector/ at this point.

Wikidata check isn't conduct by nsi-collector, but using [this script](https://github.com/osmlab/name-suggestion-index/blob/fc370cb5351283c603c415009614b19daaa3a1f5/scripts/build_wikidata.js), those who release new version package run it.

Like this: https://github.com/osmlab/name-suggestion-index/commit/72925ec297b6a3652c5c0cc21e7907ac8044e85a


> Will names that no longer exist be removed from the database?

If this “names that no longer exist” means a invalid or missing wikidata item, sure, this will cause a warning while run script 

If this “names that no longer exist” means NSI entry, no, it won't be auto removed once collected, even if there is no OSM element still using it as brand/operator/network

**Comment by Gazer75** [comment]
So basically every company name change is going to require a manual fix in the database...
I see potential for wrong suggestions happening here and iD users not knowing better will be using the wrong suggestion, which leads to bad edits...

**Comment by Snowysauce** [comment]
> Example: BKK changed its name to Eviny and retained the old BKK for its grid operator subsidiary. But initially it used the old "BKK Nett" and recently renamed it to just "BKK". (Not updated in OSM to reflect this yet). At the same time the operations of power stations is done by Eviny Fornybar, another subsidiary.

As of this writing, NSI has presets for Eviny Fornybar and BKK in the appropriate categories. Any remaining issues with Norwegian power operator presets can be raised in specific issues such as was done in #8032.
